tools/elfdeps.c

```diff
@@ -222,8 +333,17 @@ static void processDynamic(Elf_Scn *scn, GElf_Shdr *shdr, elfInfo *ei)
 	    case DT_NEEDED:
 		if (genRequires(ei)) {
 		    s = elf_strptr(ei->elf, shdr->sh_link, dyn->d_un.d_val);
-		    if (s)
-			addDep(&ei->requires, s, NULL, ei->marker);
+		    if (s) {
+			char *libtool_ver = NULL;
+			// If soname matches an item already in the deps, then
+			// it had versioned symbols and doesn't require fallback.
+			if (libtool_version_fallback &&
+			    !findSonameInDeps(ei->requires, s)) {
+			    libtool_ver = getLibtoolVerFromShLink(s);
```

Comment:
The code-block indents to the same level as the line-continued condition, which makes it unnecessarily hard to read. Pushing the condition further to right is one option, moving the opening brace to a line of its own is another.